### ISSUE TITLE
Remove the extra indentation lines for code inside the same block? 

### ISSUE THREAD

**Issue by anthony-S93** [lukas-reineke/indent-blankline.nvim#470]
Currently, the plugin always adds lines as long as the lines of code are: 1) separated by at least one newline character and 2) have the same indentation level. 

But what if I only want the lines to help me identify the scope? As in if the lines of code are in the same scope, no indent lines should appear between them **unless** the indentation level is different. 

Here are a couple of screenshots to help me explain what I mean. 
![Screenshot_2022-08-14_14-37-24](https://user-images.githubusercontent.com/69449791/184525522-087f0d3e-afd1-47dc-964d-0a4da55f43de.jpg)

Using indent-blankline, the indent lines appeared between the two `print(message)` function calls even though both calls are inside the same scope. 

With VSCode, this doesn't occur. 
![Screenshot_2022-08-14_14-39-22](https://user-images.githubusercontent.com/69449791/184525587-eaae370e-fd07-4298-9964-54c5722eab7b.jpg)
Note the lack of indent lines between the two `print(message)` function calls. It seems VSCode only adds indent lines to delineate the _scope_, which is what I want. 

How can I tweak indent-blankline's configuration to produce the same effect?

To summarize, here is the effect I want: 
![Screenshot_2022-08-14_14-45-28](https://user-images.githubusercontent.com/69449791/184525770-95f48558-9e8c-4df8-80a8-653302781bc2.jpg)
Is it possible?

**Comment by lukas-reineke** [comment]
This is an option `:help g:indent_blankline_show_trailing_blankline_indent`

**Comment by anthony-S93** [comment]
Hey, Lukas. 

Thanks for the suggestion. I have set `show_trailing_blankline_indent = true`. Here is the result. 
![Screenshot_2022-08-15_12-17-35](https://user-images.githubusercontent.com/69449791/184576248-23864feb-0515-4f7d-9188-4b4d15afc209.jpg)

The lines between the two `print(message)` did disappear, but there are still trailing indent lines after the `while` block and between the function and class definitions. Is this behavior expected?

**Comment by lukas-reineke** [comment]
Yes, the default indent algorithm does not know about semantics.

You can try `:help g:indent_blankline_use_treesitter`, which is language aware. In theory that should solve these issues, but treesitter indent is still an experimental feature. There are still a lot of cases where it makes mistakes.

**Comment by anthony-S93** [comment]
@lukas-reineke  

Thanks for your answer. Setting `indent_blankline_use_treesitter` did not fix the issue. Noted with regards to the experimental nature of treesitter. Will wait for a future fix.
